### PR TITLE
JS: Change js/misspelled-variable-name to sometimes report the local variable as the misspelling

### PR DESCRIPTION
**Pull request by erik-krogh**
Inspired by these alerts: https://lgtm.com/projects/g/MicrosoftDocs/iis-docs/alerts?id=js%2Fmisspelled-variable-name&mode=list   
(From an anomalous query result)

The `js/misspelled-variable-name` query finds a pair of a local and global variable, where one seems to be a misspelling of the other. And the query always reports the global variable as the misspelling. 

This PR adds a heuristic to check if the local variable is likely the misspelling, based on the number of occurrences of each spelling.  
If the local seems to be the misspelling, then the alert message is changed, but the alert location stays the same to avoid alerts shifting around. 

[It seems to work in practice](https://lgtm.com/query/4668972280410117144/), and gives an alert message that makes sense in more cases than previously.  
([Here are results](https://lgtm.com/query/4042208588452152747/) on a larger benchmark selection)

### CODE REVIEW

File: change-notes/1.24/analysis-javascript.md [hunk 1]
```diff
@@ -87,6 +87,7 @@
 | Identical operands (`js/redundant-operation`) | Fewer results | This query now recognizes cases where the operands change a value using ++/-- expressions. |
 | Superfluous trailing arguments (`js/superfluous-trailing-arguments`) | Fewer results | This query now recognizes cases where a function uses the `Function.arguments` value to process a variable number of parameters. |
 | Incomplete URL scheme check (`js/incomplete-url-scheme-check`) | More results | This query now recognizes more variations of URL scheme checks. |
+| Misspelled variable name (`js/misspelled-variable-name`) | Message changed | The message for this query now correctly identifies the misspelled variable in more cases. |
```
Comment (esbena) [inline change-notes/1.24/analysis-javascript.md:90]:
```suggestion
| Misspelled variable name (`js/misspelled-variable-name`) | Message changed | The message for this query now correctly identifies the misspelled variable in additional cases. |
```
:)

**Comment by erik-krogh** [comment]
> LGTM, if this scales to big-apps.slugs, and a change-note is added.

[Big-apps look good](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-js-esben.northeurope.cloudapp.azure.com_1587145977928).  

(Except TypeScript, but that also fails on master).

File: javascript/ql/src/Expressions/MisspelledVariableName.ql [hunk 2]
```diff
@@ -14,6 +14,37 @@
...
+}
+
+/**
+ * Gets the number of times a global variable with name `name` occurs in the program.
+ */
+bindingset[name]
+int globalAccesses(string name) { result = count(GlobalVarAccess acc | acc.getName() = name) }
+
+/**
+ * Holds if our heuristic says that the local variable `lvd` seems to be a misspelling of the global variable `gvd`.
```
Comment (intrigus-lgtm) [inline javascript/ql/src/Expressions/MisspelledVariableName.ql:32]: Shouldn't this be  `gva`?

Comment (erik-krogh) [reply javascript/ql/src/Expressions/MisspelledVariableName.ql]: Yes it should.

**Pull request merged**